packages/assets-controllers/src/NftDetectionController.test.ts

```diff
@@ -871,33 +1074,35 @@ type WithControllerArgs<ReturnValue> =
 async function withController<ReturnValue>(
   ...args: WithControllerArgs<ReturnValue>
 ): Promise<ReturnValue> {
-  const [{ ...rest }, fn] = args.length === 2 ? args : [{}, args[0]];
-  const { options, config } = rest;
+  const [
+    {
+      options = {},
+      config = {},
+      mockNetworkClientConfigurationsByNetworkClientId = {},
+    },
+    testFunction,
+  ] = args.length === 2 ? args : [{}, args[0]];
 
-  const getNetworkClientById = jest.fn().mockImplementation(() => {
-    return {
-      configuration: {
-        chainId: ChainId.mainnet,
-      },
-      provider: jest.fn(),
-      blockTracker: jest.fn(),
-      destroy: jest.fn(),
-    };
-  });
+  // Explicit cast used here because we know the `on____` functions are always
+  // set in the constructor.
+  const controllerEvents = {} as ControllerEvents;
+
+  const getNetworkClientById = buildMockGetNetworkClientById(
+    mockNetworkClientConfigurationsByNetworkClientId,
+  );
 
-  const nftStateChangeListeners: ((state: NftState) => void)[] = [];
-  const preferencesStateChangeListeners: ((state: PreferencesState) => void)[] =
-    [];
   const controller = new NftDetectionController(
     {
       chainId: ChainId.mainnet,
       onNftsStateChange: (listener) => {
-        nftStateChangeListeners.push(listener);
+        controllerEvents.nftsStateChange = listener;
       },
       onPreferencesStateChange: (listener) => {
-        preferencesStateChangeListeners.push(listener);
+        controllerEvents.preferencesStateChange = listener;
+      },
+      onNetworkStateChange: (listener) => {
```

Comment:
Previously we were just passing `jest.fn()` as the value for this callback, but now we are capturing the listener so that we can call it in tests, effectively firing the event.